### PR TITLE
Beatmap details rewrite

### PR DESCRIPTION
**Pull request by DrabWeb**
screen shot 2017-09-07 at 3 24 40 pm

Made more modular for future code sharing.

Also [can resize now](https://streamable.com/hpyu3).

### CODE REVIEW

**Comment by peppy** [comment]
Structurally it looks pretty good.

I did notice that the `TestCaseBeatmapDetails` step button no longer works, likely due to early return on `value == previousValue` checks in property setters.

**Comment by DrabWeb** [comment]
Seemed pretty useless to have so I just updated the visual tests with more relevant steps instead.

File: osu.Game/Screens/Select/Details/AdvancedStats.cs [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+        public BeatmapInfo Beatmap
+        {
+            get { return beatmap; }
+            set
+            {
+                if (value == beatmap) return;
+                beatmap = value;
+
+                //mania specific
+                if ((Beatmap?.Ruleset?.ID ?? 0) == 3)
```
Comment (peppy) [inline osu.Game/Screens/Select/Details/AdvancedStats.cs]: This is fine for now, but going forward we should be exposing these (and all other difficulty information below) via the `Ruleset` interface, in a similar way to `GetBeatmapStatistics`

**Comment by peppy** [comment]
I'm fine to merge this if you agree to take a look at the [one issue i pointed out](https://github.com/ppy/osu/pull/1222#discussion_r137966865) as a next step (we need to avoid adding new dependencies on ruleset IDs, as to keep modular support for other rulesets).

**Review by peppy — APPROVED** [review]

**Pull request merged**